custom_dataset_w_labels.py

```diff
@@ -0,0 +1,259 @@
+        batch_size, sequence_length, emb_dim = embs["long_term"].size()
+        skip_frames = 100
+        view_1_id = (
+            torch.randint(sequence_length - skip_frames, (batch_size,)) + skip_frames
+        )
+        view_2_id = (
+            torch.randint(sequence_length - skip_frames, (batch_size,)) + skip_frames
+        )
+
+        view_1 = byol_preds["long_term"][torch.arange(batch_size), view_1_id]
+        view_2 = embs["long_term"][torch.arange(batch_size), view_2_id]
+
+        byol_loss_long_term = (
+            1 - F.cosine_similarity(view_1, view_2.clone().detach(), dim=-1).mean()
+        )
+
+        # backprop
+        loss = 5e2 * hoa_loss + 0.5 * byol_loss_short_term + 0.5 * byol_loss_long_term
+
+        loss.backward()
+        optimizer.step()
+
+        step += 1
+        if step % log_every_step == 0:
+            writer.add_scalar("train/hoa_loss", hoa_loss.item(), step)
+            writer.add_scalar(
+                "train/byol_loss_short_term", byol_loss_short_term.item(), step
+            )
+            writer.add_scalar(
+                "train/byol_loss_long_term", byol_loss_long_term.item(), step
+            )
+            writer.add_scalar("train/total_loss", loss.item(), step)
+
+    return step
+
+def test(model, device, dataset, writer, epoch):
+    test_idx, train_idx = train_test_split(np.arange(len(dataset)), test_size=0.8, random_state=42)
+    # get embeddings
+    embeddings = compute_representations(model, dataset, device)
+    emb_keys = ['short_term', 'long_term']
+    # decode from all embeddings
+    def decode_class(keys, target, global_pool=False):
+        if len(keys) == 1:
+            emb = embeddings[keys[0]]
+        else:
+            emb = torch.cat([embeddings[key] for key in keys], dim=2)
+        emb_size = emb.size(2)
+
+        if global_pool:
+            emb = torch.mean(emb, dim=1, keepdim=True)
+
+        train_data = [emb[train_idx].reshape(-1, emb_size), target[train_idx].reshape(-1)]
+        test_data = [emb[test_idx].reshape(-1, emb_size), target[test_idx].reshape(-1)]
+        f1_score, cm = train_linear_classfier(target.max()+1, train_data, test_data, device, lr=1e-2, weight_decay=1e-4)
+        return f1_score, cm
+
+    def decode_scalar(keys, target, global_pool=False):
+        if len(keys) == 1:
+            emb = embeddings[keys[0]]
+        else:
+            emb = torch.cat([embeddings[key] for key in keys], dim=2)
+        emb_size = emb.size(2)
+
+        if global_pool:
+            emb = torch.mean(emb, dim=1, keepdim=True)
+
+        train_data = [emb[train_idx].reshape(-1, emb_size), target[train_idx].reshape(-1, 1)]
+        test_data = [emb[test_idx].reshape(-1, emb_size), target[test_idx].reshape(-1, 1)]
+        mse = train_linear_regressor(train_data, test_data, device, lr=1e-2, weight_decay=1e-4)
+        return mse
+    
+    data = next(iter(DataLoader(dataset, batch_size=len(dataset), shuffle=False)))
```

Comment:
compute inefficient change to 
```suggestion
    keys = list(dataset.annotations.keys())
    data = {}
    for batch in DataLoader(dataset, batch_size=4, shuffle=False):
      for key in keys:
        if len(data.keys()) < len(keys):
          data[key] = batch[key]
        else:
          data[key] = torch.cat([data[key], batch[key]], axis=0)
```